src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.cs

```diff
@@ -40,12 +29,8 @@ public AbstractSyntaxTreeFactoryService(SolutionServices services)
         public abstract ParseOptions TryParsePdbParseOptions(IReadOnlyDictionary<string, string> metadata);
         public abstract SyntaxTree CreateSyntaxTree(string filePath, ParseOptions options, Encoding encoding, SourceHashAlgorithm checksumAlgorithm, SyntaxNode root);
         public abstract SyntaxTree ParseSyntaxTree(string filePath, ParseOptions options, SourceText text, CancellationToken cancellationToken);
-        public abstract SyntaxTree CreateRecoverableTree(ProjectId cacheKey, string filePath, ParseOptions options, ITextAndVersionSource text, LoadTextOptions loadTextOptions, Encoding encoding, SyntaxNode root);
         public abstract SyntaxNode DeserializeNodeFrom(Stream stream, CancellationToken cancellationToken);
 
-        public virtual bool CanCreateRecoverableTree(SyntaxNode root)
-            => root.FullSpan.Length > _minimumLengthForRecoverableTree.Value;
```

Comment:
the length of a tree can't be more than int.MaxValue, so this method becamse "always false" And was inlined.